src/tfr_aruco/src/tfr_aruco_server.cpp

```diff
@@ -0,0 +1,134 @@
+        
+    // set up board. This method is temporary until an official board is created. Works for now
+    // represents the board that comes in the folder of this project
+    std::vector<std::vector<cv::Point3f> > boardCorners;
+    
+    // push top-left corners of markers
+    // This is only temporarily hardcoded
+    // TODO: write a program that generates both the code to do this from a "aruco board file"
+    // and an image that, when printed, will map directly to that code
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(0, 0, 0)));
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(6, 0, 0)));
+    
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(0, 6, 0)));
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(6, 6, 0)));
+    
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(0, 12, 0)));
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(6, 12, 0)));
+    
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(0, 18, 0)));
+    boardCorners.push_back(std::vector<cv::Point3f>(1, cv::Point3f(6, 18, 0)));
+    
+    // add rest of marker from the top left points (all the same size)
+    for(auto &marker : boardCorners) 
+    {
+        marker.push_back(marker[0] + cv::Point3f(3.2, 0,   0));
+        marker.push_back(marker[0] + cv::Point3f(3.2, 3.2, 0));
+        marker.push_back(marker[0] + cv::Point3f(0,   3.2, 0));
+        
+        // from cm to meters
+        for(auto &p : marker) {
+            p *= .01;
+        }
+    }
+    std::vector<int> boardIds = {1, 2, 3, 4, 5, 6, 7, 8};
+    
+    cv::Ptr<cv::aruco::Board> board = cv::aruco::Board::create(std::move(boardCorners), dictionary, std::move(boardIds));
+    
+    
+    // set up params
+    cv::Ptr<cv::aruco::DetectorParameters> params = cv::Ptr<cv::aruco::DetectorParameters>(new cv::aruco::DetectorParameters);
+    params->cornerRefinementMethod = cv::aruco::CORNER_REFINE_SUBPIX;
+    //params->cornerRefinementMethod = cv::aruco::CORNER_REFINE_CONTOUR;
+    params->cornerRefinementWinSize = 10;
+
+
+    image_geometry::PinholeCameraModel cameraModel;
+    cameraModel.fromCameraInfo(goal->camera_info);
+
+
+
+
+    // convert ROS message to opencv image
+    cv_bridge::CvImagePtr cv_ptr;
+    try 
+    {
+        cv_ptr = cv_bridge::toCvCopy(goal->image, sensor_msgs::image_encodings::BGR8);
+    }
+    catch (cv_bridge::Exception& e)
+    {
+        ROS_ERROR("cv_bridge exception: %s", e.what());
+        return;
+    }
+    // using cameraImage is the same as cv_ptr->image, but more convienent and clear 
+    cv::Mat cameraImage = cv_ptr->image;
+
+
+    // detect fiducial markers
+    std::vector<int> markerIds;
+    std::vector<std::vector<cv::Point2f> > markerCorners;
+    
+    cv::aruco::detectMarkers(cameraImage, dictionary, markerCorners, markerIds, params);
```

Comment:
You say that using `cameraImage` is the same as using `cv_ptr->image`, but you only use it once. You could get the same clarity by renaming `cv_ptr` to something more meaningful, like `cameraData`. Then you could use `cameraData->image` and it is just as clear.